### ISSUE TITLE
In `plugin-content-docs`, when `path` contains a dash, anything after the dash is dropped

### ISSUE THREAD

**Issue by vjpr** [facebook/docusaurus#3801]
## 🐛 Bug Report

From: https://github.com/facebook/docusaurus/issues/2528#issuecomment-731650327

```
   [
      '@docusaurus/plugin-content-docs',
      {
        id: 'foo-bar',
        path: 'foo-bar',
        routeBasePath: 'foo-bar',
      }
   ]
```

But it errors while looking inside default:
`docs/.docusaurus/docusaurus-plugin-content-docs/default/site-community-doc-1-md-87a.json`

...but the file is in community instead:

`docs/.docusaurus/docusaurus-plugin-content-docs/community/site-community-doc-1-md-87a.json`

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

## To Reproduce

(Write your steps here:)

Add plugin as follows:

```
   [
      '@docusaurus/plugin-content-docs',
      {
        id: 'foo-bar',
        path: 'foo-bar',
        routeBasePath: 'foo-bar',
      }
   ]
```

It will error when rendering this page.

## Expected behavior

No error.

(Write what you thought would happen.)

## Actual Behavior

ENOENT error.

## Your Environment

- Docusaurus version used: 2
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0):
- Operating system and version (desktop or mobile): desktop

## Reproducible Demo

TODO

**Comment by slorber** [comment]
Thanks @vjpr 
Can you give me more details about the error you see? 
You probably have more output than just ENOENT?
How have you figured this error out?


I'm able reproduce something locally, by using a - in the path of our own website, but I don't see the same error as you.

![image](https://user-images.githubusercontent.com/749374/100003338-30c62a80-2dc6-11eb-8c8f-040ff13b56cd.png)

I am unable to run `yarn start` due to this error:

![image](https://user-images.githubusercontent.com/749374/100003093-d6c56500-2dc5-11eb-83f9-7a840e1b8421.png)

But I'm able to run `yarn build`

Will try to resolve my bug, but still need to figure out a repro for yours if it's different.

**Comment by tomchen** [comment]
Is it related to #3818 that I've just posted?

I've tried some `id`, `path` and `routeBasePath` of docs, what I understand is:
- `id` is tested against `/^[a-zA-Z_\-]+$/` (letters, `-`, `_` allowed but no numbers)
- `path` can contain letters, numbers, `-`, `_`. But, there is issue #3818
- `routeBasePath` can contain letters, numbers, `-`, `_`, no issues

I too don't see `path: 'foo-bar',` could raise any problem.
